encoding/wave/decode.go

```diff
@@ -0,0 +1,217 @@
+package wave
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"reflect"
```

Comment:
i feel dirty =(

Reply:
uahuhuehuheuae

curious to why you needed it